proposals/table-component.md

```diff
@@ -0,0 +1,221 @@
+# Table Component
+
+Component to render tabular data.
+
+## Features
+
+* Sorting
+
+## Currently Unsupported
+
+* Pagination
+* Filtering
+* Fixed row
+* Fixed column
+* Striped rows
+* Hoverable row styling
+* Multi-column sort
+* Row headers
+* Expandable row
+
+## Usage
+
+### Basic usage
+
+```javascript
+import React from 'react';
+import {Table} from 'baseui/table';
+
+const data = [
+  [1, 'John', 'Smith'],
+  [2, 'Lois', 'Lane'],
+];
+
+const columns = {
+  {
+    title: 'Employee ID',
+    dataIndex: 1,
+  },
+  {
+    title: 'First',
+    dataIndex: 2,
+  },
+  {
+    title: 'Last',
+    dataIndex: 3,
+  }
+};
+
+export default () => <Table data={data} columns={columns}/>;
+```
+
+### Advanced Usage
+
+#### Sorting and custom rendering
+
+```javascript
+import React from 'react';
+import {Table, StyledCell} from 'baseui/table';
+
+const data = [
+  ['1', 'Sarah', 'Brown', 31, 'New York No. 1 Anywhere'],
+  ['2', 'Jane', 'Smith', 32, 'San Francisco No. 1 Anywhere'],
+  ['3', 'Joe', 'Black', 33, 'Sydney No. 1 Anywhere'],
+  ['4', 'Jane', 'Red', 34, 'London No. 1 Anywhere'],
+];
+
+const columns = [
+  {
```

Comment:
Missing `dataIndex`?

Reply:
my thought is that it should be optional, if a `render` function is provided. what do you think?